src/custom/state/swap/reducer.ts

```diff
@@ -77,3 +82,22 @@ export default createReducer<SwapState>(initialState, (builder) =>
       state.recipient = recipient
     })
 )
+
+/**
+ * Calculates the independent field when switching the tokens
+ *
+ * We don't support native token BUY orders
+ * So if BUY currency is native and user is inverting tokens, force independentField to INPUT
+ * to not end up with a BUY order
+ * Special case when opposite token is wrapped native, then it's fine to invert as it's a wrap
+ */
+function getIndependentField(state: SwapState): Field {
+  const chainId = state.chainId || ChainId.MAINNET
+
+  const isNativeOut =
+    state[Field.OUTPUT].currencyId?.toUpperCase() === NATIVE_CURRENCY_BUY_TOKEN[chainId].symbol?.toUpperCase()
+  const isWrappedIn =
+    state[Field.INPUT].currencyId?.toUpperCase() === WRAPPED_NATIVE_CURRENCY[chainId].symbol?.toUpperCase()
+
+  return isNativeOut && !isWrappedIn ? Field.INPUT : state.independentField === Field.INPUT ? Field.OUTPUT : Field.INPUT
+}
```

Comment:
This change addresses the edge case of a BUY order being created when switching the tokens